### PR TITLE
Add `JsonValue.TryParse`

### PR DESCRIPTION
**Pull request by mjgpy3**
Perhaps the dumbest PR so I understand if this is rejected but, that being said, here it is!

My reasons for submitting it are
 - Wrapping in option helps with an functional style, rather than having exceptions and, selfishly
 - I write this same wrapper in every project that uses FSharp.Data.

Let me know your thoughts!

### CODE REVIEW

**Comment by realvictorprm** [comment]
Actually not that worse I think, if you have some use-case for this it should be fine because it doesn't breaks anything nor it infers wrong usage.

File: tests/FSharp.Data.Tests/JsonValue.fs [hunk 1]
```diff
@@ -416,4 +426,4 @@ let ``Encoding of markup is not overzealous`` () =
   let w = new IO.StringWriter()
   JsonValue.JsonStringEncodeTo w input
   let expected = input // Should not escape </>
-  (w.GetStringBuilder().ToString()) |> should equal expected
\ No newline at end of file
+  (w.GetStringBuilder().ToString()) |> should equal expected
```
Comment (realvictorprm) [inline tests/FSharp.Data.Tests/JsonValue.fs]: Hm, could you please remove the newline afterwards? It's unnecessary here 😕

**Comment by realvictorprm** [comment]
@tpetricek sorry Sir for talking to you that much today but I struggled over this PR. Just wanted to know what you think, is this ready to merge?

**Comment by realvictorprm** [comment]
Sure, I'ld try my best @tpetricek.
I noticed that writing parsing stuff with F# is much fun and something I'm definitely continuing :smile:

File: tests/FSharp.Data.Tests/JsonValue.fs (hunk 1 above)
Comment (mjgpy3) [reply tests/FSharp.Data.Tests/JsonValue.fs]: @realvictorprm, this is done.

**Comment by mjgpy3** [comment]
@realvictorprm, @tpetricek I've squashed the commits but I can't seem to build FSharp.Data locally (even just from master). Any ideas as to why the appveyour is failing?

**Comment by realvictorprm** [comment]
@mjgpy3 the build worked for me yesterday, however I noticed that the some HttpRequest tests on the CI fail. However this is very unrelated to your addition.

**Comment by mjgpy3** [comment]
@realvictorprm I see, thank you.

**Comment by ovatsus** [comment]
I think this is fine for now, but long term we should change the parser so we don't need to throw and catch. And maybe could output in the failure case the error message stating where the parsing failed (or just the char numbers directly)?

**Comment by realvictorprm** [comment]
Hm, I could take care of this after I added the schema support

**Pull request merged**